kafka-ui-api/src/main/java/com/provectus/kafka/ui/kafka/KafkaService.java

```diff
@@ -156,17 +159,20 @@ private InternalClusterMetrics collectTopicsMetrics(Map<String,InternalTopic> to
                     .map( m -> m.values().stream().map(ClusterUtil::mapToInternalTopic).collect(Collectors.toList()));
     }
 
-    private Mono<InternalClusterMetrics> getClusterMetrics(AdminClient client) {
+    private Mono<InternalClusterMetrics> getClusterMetrics(KafkaCluster cluster, AdminClient client) {
         return ClusterUtil.toMono(client.describeCluster().nodes())
                 .flatMap(brokers ->
                     ClusterUtil.toMono(client.describeCluster().controller()).map(
                         c -> {
+                            jmxClusterUtil.fillJmxPool(c, cluster);
```

Comment:
Why do you add objects to the pool every time you load metrics? Looks like you have to initialize your pool once during the application startup. 
Or do we support dynamic cluster changes (if somebody adds or removes brokers from cluster)? If yes, how do you invalidate pool objects for removed brokers?

Reply:
I don't think we have to invalidate them and prefill on the query stage. They should be created dynamically at the first call.